component/config.go

```diff
@@ -119,6 +119,11 @@ func (t Type) String() string {
 	return t.name
 }
 
+// MarshalYAML marshals returns the Type name.
+func (t Type) MarshalYAML() (interface{}, error) {
```

Comment:
I'm surprised we use yaml in that command. I think `MarshalText` should work as well. That would be consistent with the rest of the codebase

Reply:
I just tried with `MarshalText` and it didn't work.  I can include both functions tho.

Reply:
NVM, it works if the implementation of `MarshalText` actually conforms to the interface lol